File: stacks/StacksJava/Stack.java
package stacks.StacksJava;

public class Stack {
    
    Nodo tos;

    public void apilar(char dato) {
        Nodo nuevo = new Nodo(dato);

        if(tos == null) {
            tos = nuevo;
        } else  {
            nuevo.setSiguiente(tos);
            tos = nuevo;
        }
    }
    
    public char desapilar() {
        Nodo temp = tos;
        tos = tos.getSiguiente();
        return temp.getDato();
    }
    
    public void imprimir() {
        Nodo temp = tos;
        while(temp != null) {
            System.out.println(temp.getDato());
            temp = temp.getSiguiente();
        }
    }
}

File: stacks/Stacks-Python/Stack.py
from Node import Node
class Stack:
    __tos = None
    
    def __init__(self):
        pass

    def push(self, dato):
        nuevo = Node(dato)
        if(self.__tos == None):
            self.__tos = nuevo
        else:
            nuevo.setSiguiente(self.__tos)
            self.__tos = nuevo
            
    def pop(self):
        temp = self.__tos
        self.__tos = temp.getSiguiente()
        return temp.getDato()
    
    def imprimir(self):
        temp = self.__tos
        while(temp != None):
            print(temp.getDato())
            temp = temp.getSiguiente()

        
